torchdynamo/symbolic_convert.py

```diff
@@ -298,7 +298,21 @@ def call_function(self, fn, args, kwargs):
                 self.push(ListVariable(items, **options))
             elif fn.fn is iter and args and isinstance(args[0], BaseListVariable):
                 assert not kwargs and len(args) == 1
-                self.push(ListIteratorVariable(args[0].items, **options))
+                l = args[0].items
+                # If the list contains exclusively nn.Modules, transform
+                # each module into a torchdynamo.NNModuleVariable
+                new_l = []
+                for item in l:
+                    if isinstance(item, torch.nn.Module):
+                        key = unique_id(item.__module__).replace("torch.nn.modules.", "")
```

Comment:
`key` should be based on the attribute name (which you have access to below)

Reply:
Do you mean the line `name = inst.argval` in `LOAD_ATTR`? That would get the name of the list, not the names of its internal elements, as far as I understand right now

Reply:
So the current behavior for keys is if you have:
```
self.foo(x)
```
that is mapped to the FX node:
```
call_module   "self_0.foo"   (x,)     {}
```

-----

For this PR  (lists of modules), if you have:
```
self.foo[0](x)
```
that is mapped to the FX node:
```
call_module   "linear_1"   (x,)     {}
```
(or something similar based on the name of the module containing it)

Am suggesting perhaps we name it something like "self.foo_0" for the first item.  "self.foo_1" for the second item. etc. 

The names don't actually matter, it is just for readable graphs.